src/linearmixedmodel.jl

```diff
@@ -296,23 +291,28 @@ function condVar(m::LinearMixedModel{T}) where {T}
     Array{T,3}[reshape(abs2.(ll ./ Ld) .* varest(m), (1, 1, length(Ld)))]
 end
 
-function createAL(allterms::Vector{Union{AbstractReMat{T},FeMat{T}}}) where {T}
-    k = length(allterms)
-    sz = [isa(t, ReMat) ? size(t, 2) : rank(t) for t in allterms]
-    A = AbstractMatrix{T}[]
-    L = AbstractMatrix{T}[]
-    for i = 1:k
-        for j = 1:i
-            Lij = densify(allterms[i]' * allterms[j])
-            push!(L, Lij)
-            push!(A, deepcopy(isa(Lij, BlockedSparse) ? Lij.cscmat : Lij))
+function pushALblock!(A, L, blk)
+    push!(L, blk)
+    push!(A, deepcopy(isa(blk, BlockedSparse) ? blk.cscmat : blk))
```

Comment:
Since all the calls to `pushALblock!` have the `blk` wrapped in `densify`, I wonder if it might make sense to have a `copy` option in that instead of `deepcopy` here. For matrices that are made dense, we're already dealing with a deep copy and so we add an extra copy here. For non dense matrices, we still need the copy of course.